dvc/exceptions.py

```diff
@@ -211,7 +211,9 @@ def __init__(self):
 
 
 class StageFileCorruptedError(DvcException):
-    def __init__(self, path):
+    def __init__(self, path, repository_path=None):
+        if repository_path and os.path.isabs(path) and repository_path in path:
+            path = os.path.relpath(path, repository_path)
```

Comment:
why repository path though? I would expect it to use relpath relative to the cwd, same as we do in all other places :) You also do that in your test. `relpath(path)` does `relpath(path, os.getcwd())` by default.

Reply:
hmm, that not only makes much more sense, but also reduces amount of code :P